### PR TITLE
update two yaml files via query_cxx

### LINKED ISSUE #735: Test and Enhance BUFR-to-IODA YAML for the adpsfc snow and sevcrs data
## Requirements

> Ctest passed for these two yaml files

### PR DESCRIPTION
**Pull request by YoulongXia-NOAA**
## Description

Update 3 yaml files below:

bufr_snow_adpsfc.yaml => bufr_ncep_snow_adpsfc.yaml
bufr_sevcrs_v2.yaml  => bufr_ncep_sevcrs.yaml

### Issue(s) addressed

Link the issues to be closed with this PR
- fixes #735 

## Acceptance Criteria (Definition of Done)

PR reviewed and ctest passed

## Dependencies

No

## Impact

No

### CODE REVIEW

File: test/testinput/bufr_ncep_snow_adpsfc.yaml [hunk 1]
```diff
@@ -49,56 +49,56 @@ observations:
       obsdataout: "./testrun/gdas.t06z.adpsfc_snow.tm00.nc"
 
       variables:
-        - name: "dateTime@MetaData"
+        - name: "MetaData/dateTim"
```
Comment (ilianagenkova) [inline test/testinput/bufr_ncep_snow_adpsfc.yaml:52]: I think you have a typo - replace dateTim with dateTim**e**

**Review by ilianagenkova — CHANGES_REQUESTED** [review]
@YoulongXia-NOAA , there is one typo to be fixed, the rest is good, thanks!

File: test/testinput/bufr_ncep_snow_adpsfc.yaml (hunk 1 above)
Comment (YoulongXia-NOAA) [reply test/testinput/bufr_ncep_snow_adpsfc.yaml]: Thank you for your checking. Yes, it is a typo.  Now I fixed it.

File: test/testinput/bufr_ncep_read_2_dim_blocks.yaml [hunk 2]
```diff
@@ -47,50 +47,50 @@ observations:
             - "*/QCPRMS"
 
       variables:
-        - name: "originatingGeneratingCenter@MetaData"
+        - name: "MetaData/originatingGeneratingCenter"
           source: variables/originatingGeneratingCenter
           longName: "Originating Generating Center"
           units: "id"
 
-        - name: "pressure@GeneratingApplication"
```
Comment (nicholasesposito) [inline test/testinput/bufr_ncep_read_2_dim_blocks.yaml:55]: What is the "GeneratingApplication" group? It's not in the ObsSpace.yaml or the Conventions document. Are we sure it shouldn't just be the ObsValue group? This goes for variables with the "PercentConfidence" group below, too.

File: test/testinput/bufr_ncep_snow_adpsfc.yaml [hunk 3]
```diff
@@ -26,7 +26,7 @@ observations:
             query:  "[*/CLAT, */CLATH]"
 
           stationPressure:
-            query: "[*/PRES]"
```
Comment (nicholasesposito) [inline test/testinput/bufr_ncep_snow_adpsfc.yaml:29]: I think you can take away the brackets

File: test/testinput/bufr_ncep_read_2_dim_blocks.yaml (hunk 2 above)
Comment (YoulongXia-NOAA) [reply test/testinput/bufr_ncep_read_2_dim_blocks.yaml:55]: @nicholasesposito, this is what we discussed before for this yaml file. I am not sure if Greg can accept them. @emilyhcliu perhaps help me to make an issue to discuss. Right now I just leave them there.

File: test/testinput/bufr_ncep_sevcsr.yaml [hunk 4]
```diff
@@ -1,4 +1,4 @@
-# (C) Copyright 2021 UCAR1:wq
```
[the remark this answers is not in the document]
Comment (YoulongXia-NOAA) [reply test/testinput/bufr_ncep_sevcsr.yaml:1]: this the old update. If you see new one. I has been fixed.

File: test/testinput/bufr_ncep_snow_adpsfc.yaml (hunk 3 above)
Comment (YoulongXia-NOAA) [reply test/testinput/bufr_ncep_snow_adpsfc.yaml:29]: Yes, fixed.

File: test/testinput/bufr_ncep_read_2_dim_blocks.yaml (hunk 2 above)
Comment (YoulongXia-NOAA) [reply test/testinput/bufr_ncep_read_2_dim_blocks.yaml:55]: @nicholasesposito, if possible, I am able to bring such a yaml file to JEDI IODA Hour and see what other people will recommend.  If you have any thoughts, we can discuss during that meeting

**Comment by emilyhcliu** [comment]
I checked out the branch.  I am running and testing the branch now!

File: test/testinput/bufr_ncep_read_2_dim_blocks.yaml (hunk 2 above)
Comment (rmclaren) [reply test/testinput/bufr_ncep_read_2_dim_blocks.yaml:55]: This YAML file is a unit test that tests the reading of 2 dimensional blocks... The underlying data (and the naming convetnions) are irrelevant to the purpose of this YAML file (no need to say ncep in the file name). Emily is correct.

File: test/CMakeLists.txt [hunk 5]
```diff
@@ -159,16 +159,16 @@ if( iodaconv_bufr_ENABLED )
     testinput/bufr_splitting.yaml
     testinput/bufr_filter_split.yaml
     testinput/bufr_adpsfc.yaml
-    testinput/bufr_snow_adpsfc.yaml
-    testinput/bufr_read_2_dim_blocks.yaml
+    testinput/bufr_ncep_snow_adpsfc.yaml
+    testinput/bufr_ncep_read_2_dim_blocks.yaml
```
Comment (rmclaren) [inline test/CMakeLists.txt:163]: this name change is uneccesary.

File: test/testinput/bufr_ncep_read_2_dim_blocks.yaml (hunk 2 above)
Comment (YoulongXia-NOAA) [reply test/testinput/bufr_ncep_read_2_dim_blocks.yaml:55]: @emilyhcliu, I modified this PR to remove this yaml file you suggested. Do I need to modify  CMakeLists.txt  in test directory, and copy bufr_read_2_dim_blocks.nc to and bufr_read_2_dim_blocks.yaml from feature/query_cxx to corresponding directories, git add, git comiit, and git push and make my PR clean, right?

**Review by rmclaren — CHANGES_REQUESTED** [review]

File: test/CMakeLists.txt (hunk 5 above)
Comment (YoulongXia-NOAA) [reply test/CMakeLists.txt:163]: @emilyhcliu, this branch has conflicts that must be resolved. How to resolve this conflicts? Could you give me a help?

Comment (YoulongXia-NOAA) [reply test/CMakeLists.txt]: > this name change is uneccesary.

Yes, I removed ncep and modified the corresponding CMakeLists.txt. Tested and it works. Now to push to github.

File: test/testinput/bufr_ncep_read_2_dim_blocks.yaml (hunk 2 above)
Comment (YoulongXia-NOAA) [reply test/testinput/bufr_ncep_read_2_dim_blocks.yaml]: > @YoulongXia-NOAA This YAML is related to satellite winds. This one is being taken care of in the ioda-converter code sprint. I will take care of this. You can remove this YAML from the PR.

@emilyhcliu , Greg only modify dateTime and units. He did not take care of varibleName**@**somting to something**/**variableName. You may need to modify them. This is what I found. Please double check.

**Comment by YoulongXia-NOAA** [comment]
@emilyhcliu, this validator result and it looks reasonable:  /scratch1/NCEPDEV/da/Youlong.Xia/test/ioda-bundle/buildSnowCxx/bin/ioda-validate.x    /scratch1/NCEPDEV/da/Youlong.Xia/test/ioda-bundle//ioda/share/ioda/yaml/validation/ObsSpace.yaml /scratch1/NCEPDEV                                                         /da/Youlong.Xia/test/ioda-bundle/buildSnowCxx/iodaconv/test/testrun/gdas.t06z.adpsfc_snow.tm00.nc                                                                                                             
Reading YAML from /scratch1/NCEPDEV/da/Youlong.Xia/test/ioda-bundle/ioda/share/ioda/yaml/validation/ObsSpace.yaml                                                                                                             
Processing data file: /scratch1/NCEPDEV/da/Youlong.Xia/test/ioda-bundle/buildSnowCxx/iodaconv/test/testrun/gdas.t06z.adpsfc_snow.tm00.nc                                                                                                             
 Verifying that all required groups exist
 Verifying group MetaData
 Verifying group ObsValue
 Verifying group /
 Verifying dimension names
 Verifying variable information
  Variable MetaData/dateTime
  Variable MetaData/stationIdentification
  Variable MetaData/latitude
  Variable MetaData/longitude
  Variable MetaData/stationElevation
  Variable ObsValue/stationPressure
  Variable ObsValue/airTemperature
  Variable ObsValue/totalSnowDepth
  Variable ObsValue/dewpointTemperature
Final results:
  # errors:      0
  # warnings:    0

CTEST_FULL_OUTPUT

**Comment by YoulongXia-NOAA** [comment]
@emilyhcliu, for ioda, I still used develop branch as before. If you want me to use feature/query_cxx branch, I am able to redo that.

**Comment by emilyhcliu** [comment]
You do not need to add group in the global attribute section:
Chang lin 48
```     
 - name: "MetaData/platformCommonName"
```
to
```     
 - name: "platformCommonName"
```

Same for line 52:
Change 
```
- name: "MetaData/platformLongDescription"
```
to
```
- name: "platformLongDescription"
```

**Comment by emilyhcliu** [comment]
@YoulongXia-NOAA I checked your output from adpsfc and sevcsr using ioda validator.  It passed all test if you made the changes I suggested to you in my previous comments.

**Comment by YoulongXia-NOAA** [comment]
> @YoulongXia-NOAA I checked your output from adpsfc and sevcsr using ioda validator. It passed all test if you made the changes I suggested to you in my previous comments.

Emily, I checked /scratch1/NCEPDEV/da/Youlong.Xia/test/ioda-bundle/buildSnowCxx/iodaconv/test/testrun/gdas.t00z.sevcsr.tm00.nc, It shows two warning message below:

 Variable ObsValue/cloudAmount
   Warning: Variable 'ObsValue/cloudAmount' has units of '0.01 1'. The YAML-specified units are '1'. Although convertible, these units are not equivalent.
  Variable ObsValue/brightnessTemperature
  Variable ObsValue/brightnessTemperatureStdDev
   Warning: Variable 'ObsValue/brightnessTemperatureStdDev' is not listed in the YAML conventions file.
Final results:
  # errors:      0
  # warnings:    2

**Comment by YoulongXia-NOAA** [comment]
> > @YoulongXia-NOAA I checked your output from adpsfc and sevcsr using ioda validator. It passed all test if you made the changes I suggested to you in my previous comments.
> 
> Emily, I checked /scratch1/NCEPDEV/da/Youlong.Xia/test/ioda-bundle/buildSnowCxx/iodaconv/test/testrun/gdas.t00z.sevcsr.tm00.nc, It shows two warning message below:
> 
> Variable ObsValue/cloudAmount Warning: Variable 'ObsValue/cloudAmount' has units of '0.01 1'. The YAML-specified units are '1'. Although convertible, these units are not equivalent. Variable ObsValue/brightnessTemperature Variable ObsValue/brightnessTemperatureStdDev Warning: Variable 'ObsValue/brightnessTemperatureStdDev' is not listed in the YAML conventions file. Final results:
> 
> # errors: 0
> # warnings: 2

I know that I used develop and you may use feature/query_cxx for ioda. Let me look if it works when I used feature/query_cxx for ioda.

**Comment by nicholasesposito** [comment]
@YoulongXia-NOAA Regarding the cloudAmount units, I believe they want us to use use `units: "1"` with "1" taking the place of "percent." At least, that's what I did to make that warning go away.

**Comment by YoulongXia-NOAA** [comment]
> @YoulongXia-NOAA Regarding the cloudAmount units, I believe they want us to use use `units: "1"` with "1" taking the place of "percent." At least, that's what I did to make that warning go away.

@Nick, could you please tell me if I need to use scale to convert percentage into 0-1? If yes, please give me an example I can use. Thank you.

**Comment by emilyhcliu** [comment]
I updated bufr_read_2_dim_blocks.yaml and the associated output file(bufr_read_2_dim_blocks.nc).  This is the YAML for satwind.

I ran ioda validator and all test passed!
```
${iodadir}/build/bin/ioda-validate.x  $yaml ${iodafile} 
+ /scratch1/NCEPDEV/da/Emily.Liu/JEDI/JEDI-ioda-query-cxx/ioda-bundle-snow/build/bin/ioda-validate.x /scratch1/NCEPDEV/da/Emily.Liu/JEDI/JEDI-ioda-query-cxx/ioda-bundle-snow/ioda/share/ioda/yaml/validation/ObsSpace.yaml /scratch1/NCEPDEV/da/Emily.Liu/JEDI/JEDI-ioda-query-cxx/run_adpsfc_snow/testrun/2020110112/bufr_read_2_dim_blocks.nc
Reading YAML from /scratch1/NCEPDEV/da/Emily.Liu/JEDI/JEDI-ioda-query-cxx/ioda-bundle-snow/ioda/share/ioda/yaml/validation/ObsSpace.yaml
Processing data file: /scratch1/NCEPDEV/da/Emily.Liu/JEDI/JEDI-ioda-query-cxx/run_adpsfc_snow/testrun/2020110112/bufr_read_2_dim_blocks.nc
 Verifying that all required groups exist
 Verifying group MetaData
 Verifying group /
 Verifying dimension names
 Verifying variable information
  Variable MetaData/dataProviderOrigin
  Variable MetaData/windDirectionPercentConfidence
  Variable MetaData/windSpeedPercentConfidence
  Variable MetaData/pressureGeneratingApplication
  Variable MetaData/pressurePercentConfidence
  Variable MetaData/longitude
  Variable MetaData/latitude
  Variable MetaData/coldClusterTemperatureGeneratingApplication
  Variable MetaData/windSpeedGeneratingApplication
  Variable MetaData/coldClusterTemperaturePercentConfidence
  Variable MetaData/windDirectionGeneratingApplication
Final results:
  # errors:      0
  # warnings:    0

CTEST_FULL_OUTPUT
```

The ctest(test_iodaconv_bufr_read_2_dim_blocks) passed as well.

**Comment by emilyhcliu** [comment]
> > @YoulongXia-NOAA I checked your output from adpsfc and sevcsr using ioda validator. It passed all test if you made the changes I suggested to you in my previous comments.
> 
> Emily, I checked /scratch1/NCEPDEV/da/Youlong.Xia/test/ioda-bundle/buildSnowCxx/iodaconv/test/testrun/gdas.t00z.sevcsr.tm00.nc, It shows two warning message below:
> 
> Variable ObsValue/cloudAmount Warning: Variable 'ObsValue/cloudAmount' has units of '0.01 1'. The YAML-specified units are '1'. Although convertible, these units are not equivalent. Variable ObsValue/brightnessTemperature Variable ObsValue/brightnessTemperatureStdDev Warning: Variable 'ObsValue/brightnessTemperatureStdDev' is not listed in the YAML conventions file. Final results:
> 
> # errors: 0
> # warnings: 2

@YoulongXia-NOAA two things:
(1) For cloudAmount:  The convention document has range from 0 to 1 for this variable.  The unit from BUFR for this variables is %.  Therefore, you will need to scale the cloudAmount from BUFR with 0.01
(2) For brightnessTemperatureStdDev: I added this entry in ObsSpace in IODA: feature/query_cxx.   In your ioda-bundle, under ioda, just checkout the feature/qudry_cxx and use the ObsSpace.yaml to run ioda validator.  All tests will be passed!

**Comment by YoulongXia-NOAA** [comment]
> I updated bufr_read_2_dim_blocks.yaml and the associated output file(bufr_read_2_dim_blocks.nc). This is the YAML for satwind.
> 
> I ran ioda validator and all test passed!
> 
> ```
> ${iodadir}/build/bin/ioda-validate.x  $yaml ${iodafile} 
> + /scratch1/NCEPDEV/da/Emily.Liu/JEDI/JEDI-ioda-query-cxx/ioda-bundle-snow/build/bin/ioda-validate.x /scratch1/NCEPDEV/da/Emily.Liu/JEDI/JEDI-ioda-query-cxx/ioda-bundle-snow/ioda/share/ioda/yaml/validation/ObsSpace.yaml /scratch1/NCEPDEV/da/Emily.Liu/JEDI/JEDI-ioda-query-cxx/run_adpsfc_snow/testrun/2020110112/bufr_read_2_dim_blocks.nc
> Reading YAML from /scratch1/NCEPDEV/da/Emily.Liu/JEDI/JEDI-ioda-query-cxx/ioda-bundle-snow/ioda/share/ioda/yaml/validation/ObsSpace.yaml
> Processing data file: /scratch1/NCEPDEV/da/Emily.Liu/JEDI/JEDI-ioda-query-cxx/run_adpsfc_snow/testrun/2020110112/bufr_read_2_dim_blocks.nc
>  Verifying that all required groups exist
>  Verifying group MetaData
>  Verifying group /
>  Verifying dimension names
>  Verifying variable information
>   Variable MetaData/dataProviderOrigin
>   Variable MetaData/windDirectionPercentConfidence
>   Variable MetaData/windSpeedPercentConfidence
>   Variable MetaData/pressureGeneratingApplication
>   Variable MetaData/pressurePercentConfidence
>   Variable MetaData/longitude
>   Variable MetaData/latitude
>   Variable MetaData/coldClusterTemperatureGeneratingApplication
>   Variable MetaData/windSpeedGeneratingApplication
>   Variable MetaData/coldClusterTemperaturePercentConfidence
>   Variable MetaData/windDirectionGeneratingApplication
> Final results:
>   # errors:      0
>   # warnings:    0
> 
> CTEST_FULL_OUTPUT
> ```
> 
> The ctest(test_iodaconv_bufr_read_2_dim_blocks) passed as well.

@emilyhcliu, please

**Comment by YoulongXia-NOAA** [comment]
> > > @YoulongXia-NOAA I checked your output from adpsfc and sevcsr using ioda validator. It passed all test if you made the changes I suggested to you in my previous comments.
> > 
> > 
> > Emily, I checked /scratch1/NCEPDEV/da/Youlong.Xia/test/ioda-bundle/buildSnowCxx/iodaconv/test/testrun/gdas.t00z.sevcsr.tm00.nc, It shows two warning message below:
> > Variable ObsValue/cloudAmount Warning: Variable 'ObsValue/cloudAmount' has units of '0.01 1'. The YAML-specified units are '1'. Although convertible, these units are not equivalent. Variable ObsValue/brightnessTemperature Variable ObsValue/brightnessTemperatureStdDev Warning: Variable 'ObsValue/brightnessTemperatureStdDev' is not listed in the YAML conventions file. Final results:
> > # errors: 0
> > # warnings: 2
> 
> @YoulongXia-NOAA two things: (1) For cloudAmount: The convention document has range from 0 to 1 for this variable. The unit from BUFR for this variables is %. Therefore, you will need to scale the cloudAmount from BUFR with 0.01 (2) For brightnessTemperatureStdDev: I added this entry in ObsSpace in IODA: feature/query_cxx. In your ioda-bundle, under ioda, just checkout the feature/qudry_cxx and use the ObsSpace.yaml to run ioda validator. All tests will be passed!

@Emily, please modify brightnessTemperatureStdDev as brightnessTemperatureStandardDeviation, as StdDev is abbreviation.

**Comment by YoulongXia-NOAA** [comment]
@emilyhcliu, I have made the corresponding modification for sevcsr converter based your recent comments, hopefully this is what expected. If you modify brightnessTemperatureStdDev as brightnessTemperatureStandardDeviation for ObsSpace.yaml, please test my output update. Thank you.

**Comment by emilyhcliu** [comment]
@YoulongXia-NOAA I updated the ObsSpace.yaml so the StdDev  is now StandardDeviation
There are three tests in this PR:
(1) test_iodaconv_bufr_ncep_snowadpsfc2ioda
(2)test_iodaconv_bufr_read_2_dim_blocks
(3) test_iodaconv_bufr_ncep_sevcsr

They passed unit tests (ctests) and also the ioda validator test.

**Review by emilyhcliu — APPROVED** [review]

File: test/testinput/bufr_ncep_sevcsr.yaml [hunk 6]
```diff
@@ -92,7 +92,7 @@ observations:
           coordinates: "longitude latitude Channel"
           source: variables/cloudAmount
           longName: "Amount of cloud coverage"
-          units: "%"
+          units: "percent"
           range: [0, 100]
```
[the remark this answers is not in the document]
Comment (YoulongXia-NOAA) [reply test/testinput/bufr_ncep_sevcsr.yaml]: Scale issue has been done in 9de27ab (1st update today) and units and range are done in 07acb00 (2nd update today). The issue has been solved.

**Comment by YoulongXia-NOAA** [comment]
@emilyhcliu, this what I got: 
 Warning: Variable 'ObsValue/cloudCoverTotal' does not have match any of the recommended dimensions. Variable dimensions: [ Location Channel ]. Recommended dimensions: [ Location ]. As we talked, you need to update ObsSpace.yaml in ioda.

**Comment by YoulongXia-NOAA** [comment]
@emilyhcliu, if you update this dimension, please also let Greg know. The suggestion is to open an issue for all cloudAmount, cloudcovertotal or cloudCoverageTotal including channel, etc. As cloudAmount has location and layer, so that using cloudAmount has no warnings message.  For cloud issues  found from today meeting, we need to discuss with Greg and ioda-people to find a common solution.

**Comment by emilyhcliu** [comment]
@YoulongXia-NOAA In ObsSpace.yaml.  cloudAmount and cloudCoverTotal have the following entries in the file:
```
  - Variable: [ "cloudCoverTotal" ]
    Attributes: { units: "1" }
  - Variable: [ "cloudAmount" ]
    Dimensions: [ [ "Location" ],  [ "Location", "Layer" ], [ "Location", "Channel"] ]
    Attributes: { units: "1" }
```

Since cloudAmount already has optional [location, channel] as dimension, let's use cloudAmount.  It is OK.  

I updated the output severi file.  This PR is ready for merging back to feature/query_cxx.

**Comment by YoulongXia-NOAA** [comment]
Thank you @emilyhcliu @rmclaren. Well done done.

**Pull request merged**